src/donationswap.py

```diff
@@ -318,16 +318,148 @@ def _delete_expired_offers(self):
+				html=util.Template('match-unconfirmer-email.html').replace(old_replacements).content,
+				to=old_offer.email)
+
 	def _delete_unconfirmed_matches(self):
-		return 0 #xxx not confirmed after 72 hours
+		return 0 #TODO: needs testing
+
+		count = 0
+		three_days_ago = datetime.datetime.utcnow() - datetime.timedelta(days=3)
+
+		with self._database.connect() as db:
+			for match in entities.Match.get_unconfirmed_matches(db):
+				if (match.created_ts < three_days_ago):
+					logging.info('Deleting unconfirmed match %s', match.id)
+					# match.delete(db) #TODO: check workflow with marc
+					eventlog.match_unconfirmed(db, match)
+					self._send_mail_about_unconfirmed_match(match)
+					count += 1
+
+		return count
+
+	def _send_feedback_email(self, match, db):
+		new_offer = entities.Offer.by_id(match.new_offer_id)
+		old_offer = entities.Offer.by_id(match.old_offer_id)
+
+		new_actual_amount, old_actual_amount = self._get_actual_amounts(match, new_offer, old_offer, db)
+
+		new_replacements = {
+			'{%NAME%}': new_offer.name,
+			'{%NAME_OTHER%}': old_offer.name,
+			'{%AMOUNT%}': new_actual_amount,
+			'{%CURRENCY%}': new_offer.country.currency.iso,
+			'{%CHARITY%}': new_offer.charity.name,
+			'{%AMOUNT_OTHER%}': old_actual_amount,
+			'{%CURRENCY_OTHER%}': old_offer.country.currency.iso,
+			'{%CHARITY_OTHER%}': old_offer.charity.name,
+			'{%OFFER_SECRET%}': urllib.parse.quote(new_offer.secret)
+		}
+
+		old_replacements = {
+			'{%NAME%}': old_offer.name,
+			'{%NAME_OTHER%}': new_offer.name,
+			'{%AMOUNT%}':old_actual_amount,
+			'{%CURRENCY%}': old_offer.country.currency.iso,
+			'{%CHARITY%}': old_offer.charity.name,
+			'{%AMOUNT_OTHER%}': new_actual_amount,
+			'{%CURRENCY_OTHER%}': new_offer.country.currency.iso,
+			'{%CHARITY_OTHER%}': new_offer.charity.name,
+			'{%OFFER_SECRET%}': urllib.parse.quote(old_offer.secret)
+		}
+
+		self._mail.send(
+			util.Template('email-subjects.json').json('feedback-email'),
+			util.Template('feedback-email.txt').replace(new_replacements).content,
+			html=util.Template('feedback-email.html').replace(new_replacements).content,
+			to=new_offer.email)
+
+		self._mail.send(
+			util.Template('email-subjects.json').json('feedback-email'),
+			util.Template('feedback-email.txt').replace(old_replacements).content,
+			html=util.Template('feedback-email.html').replace(old_replacements).content,
+			to=old_offer.email)
 
 	def _delete_expired_matches(self):
 		'''Send a feedback email one month after creation.'''
-		#xxx add "feedback_ts" column
-		#xxx send email to matches older than 4 weeks with empty "feedback_ts"
-		#xxx update feedback_ts
-		#xxx delete two offers and one match one week after feedback_ts
-		return 0 #xxx
+		count = 0
+		one_month_ago = datetime.datetime.utcnow() - datetime.timedelta(days=31)
+
+		with self._database.connect() as db:
+			for match in entities.Match.get_feedback_ready_matches(db):
+				if (match.created_ts < one_month_ago):
```

Comment:
feedback email is sent to people after one month
(triggered every 6 hours by memory - housekeeping cronjob)

Reply:
I see now - I thought they were part of the same work. No problem - none of the unconfirmed match email code will run because of the `return 0` in `_delete_unconfirmed_matches` so it's fine by me.

On my fork, I have created some changes for running the website locally. I think I will try to get that working this week and then we can test `_delete_unconfirmed_matches` properly before we deploy it.